widget/buffer.go

```diff
@@ -35,25 +33,26 @@ func (e *editBuffer) Changed() bool {
 	return c
 }
 
-func (e *editBuffer) deleteRunes(runes int) {
-	e.moveGap(0)
+func (e *editBuffer) deleteRunes(caret, runes int) int {
```

Comment:
I don't think that's true?  First, recall that `runes` is signed, and that `caret` only changes if `runes` < 0.  Second, `deleteRunes` deletes runes (of course), but `caret` is a byte offset.  So if we don't return the new `caret`, then in the caller we have to recalculate the new offset based on (the new value of) `e.rr.text[e.rr.gapstart]` and the following `abs(runes)` runes.

I don't think we can use `e.rr.gapend - e.rr.gapstart` because I don't see any guarantee that the gap is exactly 0 after calling moveGap, only that it starts at `caret`.

Reply:
You're absolutely right. Sorry for not thinking this through.